From 50b33903531bba256683415f4b8911c7988dd628 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 12 Jun 2024 08:51:40 +0300
Subject: [PATCH] [FSW-418] fs1: bootrom: add api to wait for tpm resp

This patch adds a function that waits for a TPM
response to become available. This function will
be used to check the availability of a response
after sending a command to TPM. With its short
timeout, it can be invoked periodically while
other ROM tasks continue to progress.

Change-Id: I3dac24da573a0d7278c1392fd515e302988cd6ed
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/common/tpm2.c b/common/tpm2.c
index d5c03e6..9b5c626 100644
--- a/common/tpm2.c
+++ b/common/tpm2.c
@@ -595,6 +595,18 @@
 	return tpm_return_code(response);
 }
 
+bool tpm2_resp_avail(bool long_timeout_en)
+{
+	int ret = 0;
+	uint8_t status;
+	uint32_t timeout;
+
+	timeout = long_timeout_en ? MS_TO_US(TPM_LONG_TIMEOUT_MS) : TPM_SHORT_TIMEOUT_US;
+	ret = tpm_spi_wait_for_stat(TPM_STS_DATA_AVAIL | TPM_STS_VALID,
+					 timeout, &status);
+	return (ret == 0);
+}
+
 #define TPM_STARTUP_COMMAND_BUF_SIZE	12
 int tpm2_startup_send_cmd(enum tpm2_startup_types mode)
 {
diff --git a/common/tpm2.h b/common/tpm2.h
index 2425c94..b970d7d 100644
--- a/common/tpm2.h
+++ b/common/tpm2.h
@@ -14,5 +14,6 @@
 int tpm2_pcr_extend_send_cmd(uint32_t index, uint32_t algorithm,
 			const uint8_t *digest, uint32_t digest_len);
 int tpm2_pcr_extend_recv_resp(void);
+bool tpm2_resp_avail(bool long_timeout_en);
 
 #endif /* _TPM2_H_ */
